### PR TITLE
BG-1341 Widget adaptations

### PR DESCRIPTION
**Pull request by SovereignAndrey**
## Explanation of the solution
Widget/Donations logic changes needed per Chauncey:
- [x] hide direct/SF breakdown lines on summary if SF portion is == 0
- [x] add SF explainer and Laira character to summary view if SF portion > 0
- [x] allow Split & BG Tip screen to be visited by Stocks and DAF for a more uniform donor flow
- [x] if `tip & SF portion === 0`, hide top total line item

### Notes:
- Need to replace the "waving" with "pointing" Laira character image, but can't find it. Will swap in when we have it. Noted a TODO comment in code addressing this.

## Instructions on making this work

- run `yarn` or `yarn install` to install npm dependencies
- run `yarn run test --watchAll` to verify all tests still pass
- (optional) run `yarn run build` to verify the build passes
- run `yarn start` to start the webapp
-

### No Tip && No SF
no-tip-no-sf

### Yes Tip && No SF
yes-tip-no-sf

### No Tip && Yes SF
no-tip-yes-sf

### Yes Tip && Yes SF
yes-tip-yes-sf

### CODE REVIEW

File: src/components/donation/Steps/common/Summary.tsx [hunk 1]
```diff
@@ -51,29 +54,43 @@ export default function Summary({
...
-        <div className="flex items-center justify-between col-span-full">
-          <dt className="mr-auto text-sm">Sustainability Fund</dt>
-          <Amount classes="text-sm" amount={locked} />
-        </div>
-        <div className="flex items-center justify-between col-span-full">
-          <dt className="mr-auto text-sm">Direct Donation</dt>
-          <Amount classes="text-sm" amount={liq} />
-        </div>
-        {props.tip && (
+        {locked > 0 && (
```
Comment (ap-justin) [inline src/components/donation/Steps/common/Summary.tsx:57]: comment: 
for me, summary is a step that lays out prior choices I've made 
e.g. opting for `100%` liquid  and `0%` locked and it's justifiable that this choice is presented to me again (summary)

image

compared to (the intent of donating is lost) and it's just assumed that user remembers that this donation is all-in liquid
image

File: src/components/donation/Steps/common/Summary.tsx [hunk 2]
```diff
@@ -51,29 +54,43 @@ export default function Summary({
...
+            </div>
+          </>
+        )}
+        {props.tip && props.tip.value > 0 && (
           <div className="col-span-full grid grid-cols-[1fr_auto] border-y border-gray-l4 py-3">
             <dt className="mr-auto">Donation for Better Giving</dt>
             <Amount classes="text-sm" amount={props.tip.value} />
           </div>
         )}
-        {props.tip && (
```
Comment (ap-justin) [inline src/components/donation/Steps/common/Summary.tsx:68]: the reason a separate `total` row is shown only when there's tip is that collation if only sensible if you have two distinct rows (og & bg)

image

so for donations that doesn't have tip, the UI is redundant
image

File: src/components/donation/Steps/common/Summary.tsx (hunk 1 above)
Comment (SovereignAndrey) [reply src/components/donation/Steps/common/Summary.tsx:57]: This is a fair point.

File: src/components/donation/Steps/common/Summary.tsx (hunk 2 above)
Comment (SovereignAndrey) [reply src/components/donation/Steps/common/Summary.tsx:68]: Per your other comment, if we always show split breakdown then this scenario can'tt happen. I think this is prob what we can do as the best balance of things.

Comment (ap-justin) [reply src/components/donation/Steps/common/Summary.tsx:68]: redundant `Total` is unncessary imo as there's only one item (even if that item's detail/partition is enumerated)
image

Comment (SovereignAndrey) [reply src/components/donation/Steps/common/Summary.tsx:68]: It's a balance between not showing the breakdown or having it, but having no tip means it's restating the first line. It's like buying a VAT free product in the store. The subtotal and total are the same numbers. One thing the second total line adds is the frequency with which the donation will occur (monthly/one-time). 

I think we can freeze this PR scope for now for review/merge and refine further, only once we know what other chagnes will be needed as a separate ticket.

Comment (ap-justin) [reply src/components/donation/Steps/common/Summary.tsx:68]: > It's like buying a VAT free product in the store. The subtotal and total are the same numbers. 

makes sense

**Pull request merged**